docker/docker-compose.yaml

```diff
@@ -0,0 +1,170 @@
+version: '3'
+services:
+    zookeeper:
+        image: wurstmeister/zookeeper
+        networks:
+            - backend
+        expose:
+            - "2181"
+        restart: unless-stopped
+    kafka:
+        image: wurstmeister/kafka:0.10.2.0
+        environment:
+            KAFKA_ADVERTISED_HOST_NAME: kafka
+            KAFKA_ADVERTISED_PORT: "9092"
+            KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+        volumes:
+            - /var/run/docker.sock:/var/run/docker.sock
+        networks:
+            - backend
+        expose:
+            - "9092"
+        depends_on:
+            - zookeeper
+        restart: unless-stopped
+    postgres:
+        image: postgres:9.6
+        environment:
+            POSTGRES_PASSWORD: $POSTGRES_PASSWORD
+        volumes:
+            - ./init-dendrite-dbs.sh:/docker-entrypoint-initdb.d/init-dendrite-dbs.sh
+            - $HOME/dendrite/postgres:/var/lib/postgresql/data
+        networks:
+            - backend
+        ports:
+            - "5432:5432"
```

Comment:
Umm, I already have a postgres installed so I'm seeing bind errors trying to start the dockerised postgres here. Would it be okay if we used a different port here to avoid the clash?

Reply:
You could treat the `docker-compose.yml` file as “production ready” or something like that, and add (development) niceties like database port forwarding to `docker-compose.override.yml` – this is an override file that is read by default by `docker-compose up`. Lastly, add `docker-compose.custon.yml` to gitignore so people can make their own overrides.